subworkflows/local/metagenomics_profiling.nf

```diff
@@ -0,0 +1,160 @@
+//
+// Complexity filtering and metagenomics screening of sequencing reads
+//
+
+// Much taken from nf-core/taxprofile subworkflows/local/profiling.nf
+
+include { MALT_RUN                       } from '../../modules/nf-core/malt/run/main'
+include { KRAKEN2_KRAKEN2                } from '../../modules/nf-core/kraken2/kraken2/main'
+include { KRAKENUNIQ_PRELOADEDKRAKENUNIQ } from '../../modules/nf-core/krakenuniq/preloadedkrakenuniq/main'
+include { METAPHLAN_METAPHLAN            } from '../../modules/nf-core/metaphlan/metaphlan/main'
+
+workflow METAGENOMICS_PROFILING {
+
+    take:
+    reads
+    database
+
+    main:
+
+    ch_versions                  = Channel.empty()
+    ch_raw_classifications       = Channel.empty()
+    ch_raw_profiles              = Channel.empty()
+    ch_multiqc_files             = Channel.empty()
+    ch_postprocessing_input      = Channel.empty()
+
+    /*
+        PREPARE PROFILER INPUT CHANNELS & RUN PROFILING
+    */
+
+    // Each tool has a slightly different input structure and generally separate
+    // input channels for reads vs database. We restructure the channel tuple
+    // for each tool and make liberal use of multiMap to keep reads/database
+    // channel element order in sync with each other
+
+    if ( params.metagenomics_profiling_tool == 'malt' ) {
+
+        // Reset entire input meta for MALT to just database name,
+        // as we don't run run on a per-sample basis due to huge databases
+        // so all samples are in one run and so sample-specific metadata
+        // unnecessary. Set as database name to prevent `null` job ID and prefix.
```

Comment:
Why not just use an alternative `tag:` in the modules config that takes from another `meta` field?

Reply:
Its like this in taxprofiler, also it doesnt really matter, since we loose all sample specific information by lumping all files together

Reply:
Mm, no it's not like this in taxprofiler - it uses the `meta.db_name` field which is what I was going for :grimacing: 

But OK, I'll leave this for now - as it works